src/transformers/trainer.py

```diff
@@ -62,7 +62,12 @@ def is_tensorboard_available():
 
 
 def is_wandb_available():
-    return _has_wandb
+    if _has_wandb:
+        wandb.ensure_configured()
+        if wandb.api.api_key == None:
+            wandb.termwarn("W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.")
```

Comment:
`wandb.termwarn` prints to stdout with some additional formating:

> *wandb*: **WARNING** W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.

Happy to change this to logger.warn if you think it would be more appropriate.  Currently this would print once on startup and then anytime logging_steps was reached.  This will only happen if wandb is installed and the environment doesn't have credentials.